src/libs/Navigation/AppNavigator/createCustomStackNavigator/CustomRouter.ts

```diff
@@ -0,0 +1,87 @@
+import {NavigationState, PartialState, RouterConfigOptions, StackNavigationState, StackRouter} from '@react-navigation/native';
+import {ParamListBase} from '@react-navigation/routers';
+import NAVIGATORS from '@src/NAVIGATORS';
+import SCREENS from '@src/SCREENS';
+import type {ResponsiveStackNavigatorRouterOptions} from './types';
+
+type State = NavigationState | PartialState<NavigationState>;
+
+/**
+ * @param state - react-navigation state
+ */
+const isAtLeastOneCentralPaneNavigatorInState = (state: State): boolean => !!state.routes.find((route) => route.name === NAVIGATORS.CENTRAL_PANE_NAVIGATOR);
+
+/**
+ * @param state - react-navigation state
+ */
+const getTopMostReportIDFromRHP = (state: State): string => {
+    if (!state) {
+        return '';
+    }
+    const topmostRightPane = state.routes.filter((route) => route.name === NAVIGATORS.RIGHT_MODAL_NAVIGATOR).at(-1);
+
+    if (topmostRightPane?.state) {
+        return getTopMostReportIDFromRHP(topmostRightPane.state);
+    }
+
+    const topmostRoute = state.routes.at(-1);
+
+    if (topmostRoute?.state) {
+        return getTopMostReportIDFromRHP(topmostRoute.state);
+    }
+
+    if (topmostRoute?.params && 'reportID' in topmostRoute.params && typeof topmostRoute.params.reportID === 'string' && topmostRoute.params.reportID) {
+        return topmostRoute.params.reportID;
+    }
+
+    return '';
+};
+/**
+ * Adds report route without any specific reportID to the state.
+ * The report screen will self set proper reportID param based on the helper function findLastAccessedReport (look at ReportScreenWrapper for more info)
+ *
+ * @param state - react-navigation state
+ */
+const addCentralPaneNavigatorRoute = (state: State) => {
+    const reportID = getTopMostReportIDFromRHP(state);
+    const centralPaneNavigatorRoute = {
+        name: NAVIGATORS.CENTRAL_PANE_NAVIGATOR,
+        state: {
+            routes: [
+                {
+                    name: SCREENS.REPORT,
+                    params: {
+                        reportID,
+                    },
+                },
+            ],
+        },
+    };
+    state.routes.splice(1, 0, centralPaneNavigatorRoute);
+    // @ts-expect-error Updating read only property
+    // noinspection JSConstantReassignment
+    state.index = state.routes.length - 1; // eslint-disable-line
```

Comment:
If you have a better idea here, please share 😅

Reply:
How about:
```ts
    // eslint-disable-next-line no-param-reassign, @typescript-eslint/non-nullable-type-assertion-style
    (state.index as number) = state.routes.length - 1;
```